infra/src/dsl/dsl.ts

```diff
@@ -241,13 +255,138 @@ export class ServiceBuilder<ServiceType> {
     return this
   }
 
-  postgres(postgres?: PostgresInfo) {
-    this.serviceDef.postgres = this.withDefaults(postgres ?? {})
+  private stripPostfix(name: string): string
+  private stripPostfix(name: undefined): undefined
+  private stripPostfix(name?: string) {
+    if (!name) return
+    // Strip -worker and -job postfixes from database name
+    for (const postfix of ['-worker', '-job']) {
+      if (name.endsWith(postfix)) {
+        name = name.replace(postfix, '')
+      }
+    }
+    return name
+  }
+
+  private grantDB(
+    target?: PostgresInfo,
+    postgres?: PostgresInfo,
+    copy = false,
+  ): PostgresInfo {
+    if (copy) {
+      const targetCopy = this.grantDB({}, target, false)
+      return this.grantDB(targetCopy, postgres, false)
+    }
+    if (!target) {
+      target = {}
+    }
+    const dbExtensions = (target.extensions ?? []).concat(
+      postgres?.extensions ?? [],
+    )
+
+    merge(target, postgres)
+    merge(target, this.postgresDefaults(postgres ?? {}))
+    if (dbExtensions) target.extensions = dbExtensions
+    if (target.extensions?.length === 0) target.extensions = undefined
+
+    return target
+  }
+
+  /**
+   * To perform maintenance before deploying the main service(database migrations, etc.), create an `initContainer` (optional). It maps to a Pod specification for an [initContainer](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/).
+   * @param ic - InitContainers definitions
+   */
+  initContainer(
+    ic: Optional<InitContainers, 'envs' | 'secrets' | 'features'>,
+    withDB = false,
```

Comment:
could you document the purpose of `withDB`? I am not sure I understand it's role here.

Reply:
Documented :heavy_check_mark: 

Only currently used for recursive defaulting.